bitcoin/src/blockdata/script/instruction.rs

```diff
@@ -61,6 +61,27 @@ impl<'a> Instruction<'a> {
             Instruction::PushBytes(bytes) => ScriptBuf::reserved_len_for_slice(bytes.len()),
         }
     }
+
+    /// Reads an integer from an Instruction,
+    /// returning Some(i64) for valid opcodes or pushed bytes, otherwise None
+    pub fn read_int(&self) -> Option<i64> {
```

Comment:
Why does this return an `i64` instead of an `i32`? (I have a vague memory that we discussed that before but cannot remember the details). The issue seems to point towards using an `i32` IIUC.

Reply:
I have a bad feeling my question is going to take us down a rabbit hole, appologies in advance.

Reply:
It's a good rabbit-hole to go down.

The reason we return an `i64` here is that in a couple of contexts (well, just one -- the input to `CHECKLOCKTIMEVERIFY`) Script allows 5-byte numbers rather than 4-byte ones.

This function, like the old `read_scriptint`, only allows reading 4-byte integers.

This is actually a missing feature that caused BitVM to fork our library. (Well, to use a development branch of Steven's; "fork the library" is maybe an extreme way to describe it). We have https://github.com/BitVM/BitVM/issues/23 about this but we don't have an issue here.

Reply:
As a secondary thing, opcodes like `ADD` `SUB` etc might output 5-byte numbers, even though they can't accept them. We have a big doccomment saying "Core won't parse these so neither will we" but possibly we want to revisit that at the same time?

Reply:
As I suspected, I can't look at any of this code meaningfully without going super deep into it, which I don't want to allocate clock cycles to right now. I think for this PR we should leave it as it is (a `u64`) and we need to come back and someone has to put in the time to fix this up. I think this PR is a good step forward but we should not close the issue when this merges. I'm going to link to this discussion from the issue.

Reply:
I've removed the "close " bit from the PR description.